### ISSUE TITLE
LPDID: Setting Windows with Monthly Data

### ISSUE THREAD

**Issue by escherpf** [py-econometrics/pyfixest#750]
LPDID currently requires that date variable inputs (tname and gname) be in the format YYYYMM if one is using monthly data. This seems to cause some weirdness with setting the pre- and post-windows when one has monthly data spanning more than a year. In particular, with dates in the YYYYMM format (e.g., 202312 and 202401) the jump from  December of one calendar year to January of the next is obviously not 1. If I set my pre- and post-window to values less than (minus and plus) 12, there's no problem: the command runs. But setting them to something like plus/minus 20 (again, in terms of months) and LPDID will not run and I get an error message like the one displayed below. On the other hand, if I set it to something large enough to cover the jump from one calendar year to the next (e.g., 96) the command runs and produces coefficients for all 96 leads and lags (I have about 28 years of data).  
 
Is there something I'm missing here, or is this an known issue? Or just something that will persist until datetime formats are supported?

> 
![image](https://github.com/user-attachments/assets/52587950-d883-4a76-a412-20b89942a88d)

**Comment by s3alfisc** [comment]
Thanks for reporting - looks like another bug! What is it that you are re-running? Is it running one of etable / summary / tidy / vcov post estimation that alters the model output? Could you provide a reproducible example with the `df_het` example data set?

**Comment by escherpf** [comment]
It occurs when I run iplot (In my original message I had put "iplot" in angle brackets, and I just now noticed that doing so suppressed that bit of text...a key piece of information! Sorry about that!). 

I will come up with a reproducible example using the `df_het` data set. Thank you!

**Comment by escherpf** [comment]
```
url = "https://raw.githubusercontent.com/py-econometrics/pyfixest/master/pyfixest/did/data/df_het.csv"
df_het = pd.read_csv(url)
fit = pf.lpdid(
    df_het,
    yname="dep_var",
    idname="unit",
    tname="year",
    gname="g",
    vcov={"CRV1": "state"},
    pre_window=-20,
    post_window=20,
    att=False
)
fit.tidy().head()
fit.iplot(figsize= [1200, 400], coord_flip=False).show()
fit.tidy().head()
fit.iplot(figsize= [1200, 400], coord_flip=False).show()
```
The first `tidy()` call, produces the following output

![image](https://github.com/user-attachments/assets/97994efc-a249-4569-a183-17387c077c1d)
 
The second `tidy()` call, after running` iplot()`, produces 

![image](https://github.com/user-attachments/assets/15b66fbe-2692-418c-ae64-6d800ae87ca3)

The second `iplot()` call, generates the following error:
 

> AttributeError: 'DataFrame' object has no attribute 'dtype'"

**Comment by s3alfisc** [comment]
Thanks @escherpf , I can reproduce this. The second evaluation of 

```python
fit.tidy().head()
fit.iplot(figsize= [1200, 400], coord_flip=False).show()
```
produces 
```python
AttributeError: 'DataFrame' object has no attribute 'dtype'
```

As a matter of fact, it seems that calling `iplot()` twice seems to trigger the error: 

```python
fit = pf.lpdid(
    df_het,
    yname="dep_var",
    idname="unit",
    tname="year",
    gname="g",
    vcov={"CRV1": "state"},
    pre_window=-20,
    post_window=20,
    att=False
)
fit.iplot(figsize= [1200, 400], coord_flip=False).show()
fit.iplot(figsize= [1200, 400], coord_flip=False).show()
```
I'll take a look at this in the next days.

**Comment by escherpf** [comment]
> As a matter of fact, it seems that calling iplot() twice seems to trigger the error:

That's correct. Calling `iplot` changes the index of LPDID object, which is what appears to be causing the error on the second `iplot` call. 

Thank you @s3alfisc !

**Comment by s3alfisc** [comment]
The iplot error does not occur with `did2s`:

```python
fit = pf.did2s(
    df_het,
    yname="dep_var",
    first_stage="~ 0 | unit + year",
    second_stage="~i(rel_year, ref=-1.0)",
    treatment="treat",
    cluster="state",
)
fit.iplot(figsize= [1200, 400], coord_flip=False).show()
fit.iplot(figsize= [1200, 400], coord_flip=False).show()
```

**Comment by s3alfisc** [comment]
The issue with `iplot()` and reindexing should be solved after merging #754, was an easy fix =)

**Comment by s3alfisc** [comment]
Currently doing some research on how the leading did libraries handle missing data. 

- `did` allows only numeric input: [link](https://github.com/bcallaway11/did/blob/6ada479b3dd08cc1f1ba5bc8ec0fbd19bcbfa0f1/R/pre_process_did.R#L50)

**Comment by s3alfisc** [comment]
Btw I need to add the name of the required input data type for tname, gname, etc to the docs: 
![image](https://github.com/user-attachments/assets/b2873b3a-b88f-4892-aec6-3307442ab23d)

**Comment by s3alfisc** [comment]
Made some changes in this PR: https://github.com/py-econometrics/pyfixest/pull/756 . Could you check if these make sense for you @escherpf?

**Comment by escherpf** [comment]
Just to follow up on this issue: short of explicitly supporting datetime formats, would there be a way to tweak the API for `lpdid`, similar to that for `did2s`, where the user can provide the relative time variable directly (`rel_year` in the [code](https://github.com/py-econometrics/pyfixest/blob/master/pyfixest/did/lpdid.py))? That way, one circumvents the problem of the inputs `tname` and `gname` not producing a sensible relative time variable when these are in numeric YYYYMM (i.e., monthly) format.
